modules/social_features/social_user_export/src/Plugin/UserExportPluginBase.php

```diff
@@ -215,25 +214,21 @@ public function profileGetTaxonomyFieldValue($field_name, ProfileInterface $user
       return '';
     }
 
-    $concatenated_value = '';
-    try {
-      $values = $user_profile->get($field_name)->getValue();
-      if (is_array($values) && !empty($values)) {
-        $taxonomy_terms = [];
-        foreach ($values as $x => $value) {
-          if (isset($value['target_id'])) {
-            /** @var \Drupal\taxonomy\Entity\Term $term */
-            $taxonomy_term = Term::load($value['target_id']);
-            $taxonomy_terms[] = $taxonomy_term->getName();
-          }
-        }
-        $concatenated_value = implode(', ', $taxonomy_terms);
-      }
+    /** @var \Drupal\Core\Field\EntityReferenceFieldItemListInterface $field */
+    $field = $user_profile->get($field_name);
+
+    if ($field->isEmpty()) {
+      return '';
     }
-    catch (\Exception $e) {
-      $concatenated_value = '';
+
+    $names = [];
```

Comment:
We could use `array_map` here :)

```suggestion
    $names = array_map(
      /** @var \Drupal\taxonomy\TermInterface $taxonomy_term */
      function ($taxonomy_term) {
        return $taxonomy_term->getName();
      },
      $user_profile->get($field_name)->referencedEntities()
    );
```

`array_map` on an empty array still returns an empty array. This block allows us to remove the foreach.

It's also safe to call implode on an empty array because it returns an empty string.